### PR TITLE
Fix [FormChipCell] issues with labels

### PR DESCRIPTION
**Pull request by Taras-Hlukhovetskyi**
https://jira.iguazeng.com/browse/ML-5785

### CODE REVIEW

File: src/lib/components/FormChipCell/NewChipForm/NewChipForm.js [hunk 1]
```diff
@@ -232,6 +239,14 @@ const NewChipForm = React.forwardRef(
           }
 
           event && event.stopPropagation()
+
```
Comment (mavdryk) [inline src/lib/components/FormChipCell/NewChipForm/NewChipForm.js]: delete empty line

File: src/lib/components/FormChipCell/NewChipForm/NewChipForm.js [hunk 2]
```diff
@@ -95,6 +95,11 @@ const NewChipForm = React.forwardRef(
         'item_edited_invalid'
     )
 
+    const closeButtonClass = classnames(
+      'edit-chip__icon-close',
+      (editConfig.chipIndex === chipIndex || !isEditable) && 'edit-chip__icon-close-hidden'
```
Comment (mavdryk) [inline src/lib/components/FormChipCell/NewChipForm/NewChipForm.js]: use BEM naming.
`edit-chip__icon-close_hidden`

File: src/lib/components/FormChipCell/FormChipCell.js [hunk 3]
```diff
@@ -222,14 +225,33 @@ const FormChipCell = ({
...
+        }
+
         setEditConfig((preState) => ({
           ...preState,
           chipIndex: index,
           isEdit: true,
-          isKeyFocused: true,
-          isValueFocused: false
+          isKeyFocused: isKeyClicked ? true : false,
+          isValueFocused: isKeyClicked ? false : true
```
Comment (mavdryk) [inline src/lib/components/FormChipCell/FormChipCell.js]:
```suggestion
          isKeyFocused: isKeyClicked,
          isValueFocused: !isKeyClicked
```

File: src/lib/components/FormChipCell/FormChipCell.js [hunk 4]
```diff
@@ -189,25 +192,25 @@ const FormChipCell = ({
         }
 
         setEditConfig((prevState) => {
-          const isPrevChipIndexExists = prevState.chipIndex - 1 < 0
+          const firstChipSelected = prevState.chipIndex === 0
```
Comment (mavdryk) [inline src/lib/components/FormChipCell/FormChipCell.js]: rename to `firstChipIsSelected`

File: src/lib/components/FormChipCell/FormChipCell.js [hunk 5]
```diff
@@ -220,16 +223,46 @@ const FormChipCell = ({
   )
 
   const handleToEditMode = useCallback(
-    (event, index) => {
+    (event, chipIndex, keyName) => {
       if (isEditable) {
+        const { clientX: pointerCoordinateX, clientY: pointerCoordinateY } = event
+        let isKeyClicked = false
+        const isClickedInsideElement = (pointerCoordinateX, pointerCoordinateY, element) => {
```
Comment (mavdryk) [inline src/lib/components/FormChipCell/FormChipCell.js]: Not sure what element is? is it chip itself or input? use proper name inputElement or chipElement

File: src/lib/types.js [hunk 6]
```diff
@@ -76,7 +76,9 @@ export const CHIPS = PropTypes.arrayOf(CHIP)
 
 export const POP_UP_CUSTOM_POSITION = PropTypes.shape({
   element: PropTypes.shape({}),
-  position: PropTypes.oneOf(['top-left', 'top-right', 'bottom-left', 'bottom-right'])
+  position: PropTypes.oneOf(['top-left', 'top-right', 'bottom-left', 'bottom-right']),
+  switchOppositeHorizontal: PropTypes.bool,
+  switchOppositeVertical: PropTypes.bool
```
Comment (mavdryk) [inline src/lib/types.js]: change props to `autoHorizontalPosition` and `autoVerticalPosition`

**Review by mavdryk — CHANGES_REQUESTED** [review]

File: src/lib/components/FormChipCell/formChipCell.util.js [hunk 7]
```diff
@@ -15,3 +15,31 @@ under the Apache 2.0 license is conditioned upon your compliance with
...
+
+    input.id = 'chips-hidden-util-input'
+    input.tabIndex  = -1
+    document.body.append(input)
+  }
+
+  input.value = value
+
+  return input.scrollWidth ?? 0
+}
```
Comment (Taras-Hlukhovetskyi) [inline src/lib/components/FormChipCell/formChipCell.util.js:48]: NOTE:
I've tried other approaches and nothing worked without some bugs

File: src/lib/components/FormChipCell/formChipCell.util.js [hunk 8]
```diff
@@ -15,3 +15,32 @@ under the Apache 2.0 license is conditioned upon your compliance with
 such restriction.
 */
 export const uniquenessError = { name: 'uniqueness', label: 'Key must be unique' }
+
+export const getTextWidth = (elementWitText) => {
```
Comment (mavdryk) [inline src/lib/components/FormChipCell/formChipCell.util.js]: typo in the parameter name

File: src/lib/components/FormChipCell/formChipCell.util.js [hunk 9]
```diff
@@ -15,3 +15,32 @@ under the Apache 2.0 license is conditioned upon your compliance with
...
+  if (!elementWitText) {
+    return 0
+  }
+  let span = document.getElementById('chips-hidden-utils-helper-element')
```
Comment (mavdryk) [inline src/lib/components/FormChipCell/formChipCell.util.js]: create a variable for the `id`

Comment (mavdryk) [reply src/lib/components/FormChipCell/formChipCell.util.js]: rename `span` to `hiddenElement` and the id to `chips-hidden-element`

File: src/lib/components/FormChipCell/formChipCell.util.js [hunk 10]
```diff
@@ -15,3 +15,32 @@ under the Apache 2.0 license is conditioned upon your compliance with
...
+  if (!span) {
+    span = document.createElement('span')
+    const styles = {
+      position: 'absolute',
+      left: '-10000px',
+      top: "auto",
+      visibility: 'hidden'
+    }
+
+    for (const [key, value] of Object.entries(styles)) {
```
Comment (mavdryk) [inline src/lib/components/FormChipCell/formChipCell.util.js]: rename to `styleName` and `styleValue`

File: src/lib/components/FormChipCell/formChipCell.util.js [hunk 11]
```diff
@@ -15,3 +15,32 @@ under the Apache 2.0 license is conditioned upon your compliance with
...
+      span.style[key] = value;
+    }
+
+    span.style.font = window.getComputedStyle(elementWitText).font
+    span.id = 'chips-hidden-utils-helper-element'
+    span.tabIndex  = -1
+    document.body.append(span)
+  }
+  span.textContent = elementWitText.value
```
Comment (mavdryk) [inline src/lib/components/FormChipCell/formChipCell.util.js]: add empty line above

**Review by mavdryk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**